server/chatserver.cpp

```diff
@@ -0,0 +1,56 @@
+#include "chatserver.h"
+
+static QString getIdentifier(QWebSocket* socket)
+{
+    return QStringLiteral("%1:%2").arg(socket->peerAddress().toString(),
+                                       QString::number(socket->peerPort()));
+}
+
+ChatServer::ChatServer(quint16 port, QObject *parent)
+    : QObject{parent},
+      m_webSocketServer{new QWebSocketServer("Chat Server",
+                            QWebSocketServer::NonSecureMode,
+                            this)}
+{
+    if(m_webSocketServer->listen(QHostAddress::Any, port))
+    {
+        qInfo() << "Server listening on port " << port;
+        connect(&*m_webSocketServer, &QWebSocketServer::newConnection, this, &ChatServer::onNewConnection);
```

Comment:
why &*m_webSocketServer instead of m_webSocketServer?

Reply:
this was left after trying to use smart pointers, I will change it